site/zenodo_rdm/previewer/mirador.py

```diff
@@ -64,6 +55,46 @@ def preview(file):
         if tile_file is not None and tile_file.processor:
             tile_status = tile_file.processor.get("status")
 
+    # Check if the file was updated more than an hour ago
+    last_updated_time = datetime.fromisoformat(file.data["updated"])
+    current_time = datetime.now(timezone.utc)
+    time_diff = current_time - last_updated_time
+
+    # If the image tile status size is processing and the image was last updated more than an hour ago
+    if tile_status == "processing" and time_diff > timedelta(hours=1):
+        # The image size is less than configured size, fall back to IIIF
+        if file.size < current_app.config["PREVIEWER_MAX_IMAGE_SIZE_BYTES"]:
+            return render_template(
+                "invenio_app_rdm/records/previewers/simple_image_preview.html",
+                css_bundles=["mirador-previewer.css"],
+                file_url=file.file.links["iiif_api"],
+            )
+        # The image size is greater than configured size,
+        # image cannot be previewed
+        elif file.size > current_app.config["PREVIEWER_MAX_IMAGE_SIZE_BYTES"]:
+            return render_template(
+                "invenio_app_rdm/records/previewers/default.html",
```

Comment:
I feel like the `default` is not a great name in this case. Maybe something like?

```suggestion
                "invenio_app_rdm/records/previewers/not_available_preview.html",
```